iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ShuntCompensatorImpl.java

```diff
@@ -101,9 +101,6 @@ public ShuntCompensatorImpl setSectionCount(int sectionCount) {
         NetworkImpl n = getNetwork();
         ValidationUtil.checkSections(this, sectionCount, model.getMaximumSectionCount(), getNetwork().getMinValidationLevel(),
                 getNetwork().getReportNodeContext().getReportNode());
-        if (sectionCount < 0 || sectionCount > model.getMaximumSectionCount()) {
-            throw new ValidationException(this, "unexpected section number (" + sectionCount + "): no existing associated section");
-        }
```

Comment:
These tests are already performed in `checkSections`, the instruction before. It is therefore a dead code.